packages/meteor-wallet/src/lib/meteor-wallet.ts

```diff
@@ -0,0 +1,256 @@
+import {
+  connect,
+  keyStores,
+  transactions as nearTransactions,
+  utils,
+} from "near-api-js";
+import type {
+  WalletModuleFactory,
+  WalletBehaviourFactory,
+  InjectedWallet,
+  Transaction,
+  Optional,
+  Network,
+} from "@paras-wallet-selector/core";
+import {
+  EMeteorWalletSignInType,
+  MeteorWallet as MeteorWalletSdk,
+} from "@meteorwallet/sdk";
+import {
+  MeteorWalletParams_Injected,
+  MeteorWalletState,
+} from "./meteor-wallet-types";
+import { createAction } from "@paras-wallet-selector/wallet-utils";
+import icon from "./icon";
+
+const setupWalletState = async (
+  params: MeteorWalletParams_Injected,
+  network: Network
+): Promise<MeteorWalletState> => {
+  const keyStore = new keyStores.BrowserLocalStorageKeyStore(
+    window.localStorage,
+    "_meteor_wallet"
+  );
+
+  const near = await connect({
+    keyStore,
+    ...network,
+    headers: {},
+  });
+
+  const wallet = new MeteorWalletSdk({ near, appKeyPrefix: "near_app" });
+
+  return {
+    wallet,
+    keyStore,
+  };
+};
+
+const createMeteorWalletInjected: WalletBehaviourFactory<
+  InjectedWallet,
+  { params: MeteorWalletParams_Injected }
+> = async ({ options, logger, store, params }) => {
+  const _state = await setupWalletState(params, options.network);
+
+  const cleanup = () => {
+    _state.keyStore.clear();
+  };
+
+  const getAccounts = () => {
+    const accountId = _state.wallet.getAccountId();
+
+    if (!accountId) {
+      return [];
+    }
+
+    return [{ accountId }];
+  };
+
+  const transformTransactions = async (
+    transactions: Array<Optional<Transaction, "signerId">>
+  ) => {
+    const account = _state.wallet.account()!;
+    const { networkId, signer, provider } = account.connection;
+
+    const localKey = await signer.getPublicKey(account.accountId, networkId);
+
+    for (const trx of transactions) {
+      if (trx.signerId !== account.accountId) {
+        throw new Error(
+          `Transaction had a signerId which didn't match the currently logged in account`
+        );
+      }
+    }
```

Comment:
Hi @lostpebble, I think this part will always throw an Error if dApps doesn't include the `signerId`, since the `signerId` is optional on transactions (`transactions: Array<Optional<Transaction, "signerId">>`). 

If I check other wallets, they didn't add this checker. Probably we can omit this part or maybe do you have any suggestions?

Reply:
Ah yea, I think it was a remnant of some old wallet selector code for Near Wallet. I guess making it optional might open some functionality in the future- I've removed that check now.